client/image_tag_test.go

```diff
@@ -30,11 +30,35 @@ func TestImageTagInvalidReference(t *testing.T) {
 	}
 
 	err := client.ImageTag(context.Background(), "image_id", "aa/asdf$$^/aa")
-	if err == nil || err.Error() != `Error parsing reference: "aa/asdf$$^/aa" is not a valid repository/tag` {
+	if err == nil || err.Error() != `Error parsing reference: "aa/asdf$$^/aa" is not a valid repository/tag: invalid reference format` {
 		t.Fatalf("expected ErrReferenceInvalidFormat, got %v", err)
 	}
 }
 
+func TestImageTagInvalidSourceImageName(t *testing.T) {
+
+	client := &Client{
+		client: newMockClient(errorMock(http.StatusInternalServerError, "Server error")),
+	}
+
+	err := client.ImageTag(context.Background(), "invalid_source_image_name_", "repo:tag")
+	if err == nil || err.Error() != "Error parsing reference: \"invalid_source_image_name_\" is not a valid repository/tag: invalid reference format" {
+		t.Fatalf("expected Parsing Reference Error, got %v", err)
+	}
+}
+
+func TestImageTagInvalidSourceImageDigest(t *testing.T) {
+
+	client := &Client{
+		client: newMockClient(errorMock(http.StatusInternalServerError, "Server error")),
+	}
+
+	err := client.ImageTag(context.Background(), "nginx@sha256:fab482910aae9630c93bd24fc6fcecb9f9f792c24a8974f5e46d8ad625ac2357", "repo:tag")
```

Comment:
Can you explain this test? Having an image-digest as source should be valid (although this image won't be available in CI; the _validation_ should not fail though). 

```bash
$ docker pull busybox

Using default tag: latest
latest: Pulling from library/busybox
fdab12439263: Pull complete
Digest: sha256:f102731ae8898217038060081c205aa3a4ae3f910c2aaa7b3adeb6da9841d963
Status: Downloaded newer image for busybox:latest

$ docker tag busybox@sha256:f102731ae8898217038060081c205aa3a4ae3f910c2aaa7b3adeb6da9841d963 foo:bar

$ docker images

REPOSITORY          TAG                 IMAGE ID            CREATED             SIZE
foo                 bar                 1efc1d465fd6        2 weeks ago         1.1 MB
busybox             latest              1efc1d465fd6        2 weeks ago         1.1 MB
```